examples/optionals/as_u64.rs

```diff
@@ -0,0 +1,98 @@
+#![deny(warnings)]
+#![warn(rust_2018_idioms)]
+
+#[macro_use]
+extern crate log;
+
+use serde::Deserialize;
+use serde_this_or_that::as_opt_u64;
+
+#[derive(Clone, Debug, Deserialize)]
+pub struct Msg {
+    #[serde(deserialize_with = "as_opt_u64")]
+    pub timestamp: Option<u64>,
+}
+
+// A simple type alias so as to DRY.
+type Result<T> = std::result::Result<T, Box<dyn std::error::Error>>;
+
+fn main() -> Result<()> {
+    sensible_env_logger::init!();
+
+    trace!("With Empty String:");
+    let data = r#"
+    {
+        "timestamp": ""
+    }"#;
+
+    let m: Msg = serde_json::from_str(data).unwrap();
+    assert_eq!(m.timestamp, None);
+    trace!("  {m:?}");
+
+    trace!("With Boolean:");
+    let data = r#"
+    {
+        "timestamp": true
+    }"#;
+
+    let m: Msg = serde_json::from_str(data).unwrap();
+    assert_eq!(m.timestamp, None);
+    trace!("  {m:?}");
+
+    trace!("With Null:  ");
+    let data = r#"
+    {
+        "timestamp": null
+    }"#;
+
+    let m: Msg = serde_json::from_str(data).unwrap();
+    assert_eq!(m.timestamp, None);
+    trace!("  {m:?}");
+
+    trace!("With U64:  ");
+
+    let data = r#"
+    {
+        "timestamp": 123456789076543210
+    }"#;
+
+    let m: Msg = serde_json::from_str(data).unwrap();
+    trace!("  {m:?}");
+
+    trace!("With F64:");
+
+    let data = r#"
+    {
+        "timestamp": 0.5
+    }"#;
+
+    let m: Msg = serde_json::from_str(data).unwrap();
+    trace!("  {m:?}");
+    assert_eq!(m.timestamp, Some(1));
```

Comment:
I think this could arguably be `None`. To round or not to round...

Reply:
Agreed! When I searched online, I found that the rounding logic can be considerably more logic, as it relies on if a number to the left of the decimal point is even or odd. This makes rounding `0.5` ambiguous, as `0` is neither odd or even (or maybe both?).

For now, I wouldn't say that it's necessarily *wrong* to round 0.5 to 1. That is, it appears that the rounding logic is at least consistent, so that it is expected that 0.5 would round upwards to 1 (same with 3.5 or 4.5 for example). 

However, I'm definitely open to exploring this more, and seeing what others thoughts and opinions might arise on this issue with rounding numbers in this scenario.